src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,228 @@
+    address public constant CUSDC = address(0x39AA39c021dfbaE8faC545936693aC917d5E7563);
+    address public constant WBTC = address(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599);
+    address public constant CWBTC = address(0xC11b1268C1A384e55C48c2391d8d480264A3A7F4);
+    mapping(address => uint16) public currencyIds;
+    mapping(address => bool) public ctokens;
+    mapping(address => address) public underlyingAddrMap;
+
+    constructor(address _rollupProcessor, address _fcashFactory) BridgeBase(_rollupProcessor) {
+        fcashFactory = IWrappedfCashFactory(_fcashFactory);
+        currencyIds[ETH] = 1;
+        currencyIds[CETH] = 1;
+        currencyIds[WETH] = 1;
+        currencyIds[DAI] = 2;
+        currencyIds[CDAI] = 2;
+        currencyIds[USDC] = 3;
+        currencyIds[CUSDC] = 3;
+        currencyIds[WBTC] = 4;
+        currencyIds[CWBTC] = 4;
+        underlyingAddrMap[CETH] = WETH;
+        underlyingAddrMap[CDAI] = DAI;
+        underlyingAddrMap[CUSDC] = USDC;
+        underlyingAddrMap[CWBTC] = WBTC;
+        ctokens[CETH] = true;
+        ctokens[CDAI] = true;
+        ctokens[CUSDC] = true;
+        ctokens[CWBTC] = true;
+    }
+
+    receive() external payable {}
+
+    /**
+     * @notice Convert function called by rollup processor, will enter or exit Notional lending position
+     * @dev Will revert if the input asset or the output asset is invalid
+     * @param _inputAssetA The input asset. For entering the market, we support ETH, USDC, DAI, WBTC
+     * and their corresponding ctokens. For exiting the market, we support every valid fcash token.
+     * @param _outputAssetA The output asset. For entering the market, it should be a valid fcash token.
+     For exiting the market, this should be one of ETH, USDC, DAI, WBTC, and their corresponding ctokesn
+     * @param _inputValue The amount of input token
+     * @param _interactionNonce The interaction nonce of the call
+     * @param _auxData For entering the market, this will be the market maturity. For exitng the market,
+     * this will be 0.
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory,
+        uint256 _inputValue,
+        uint256 _interactionNonce,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        uint40 maturity = uint40(_auxData);
+        _validateInput(maturity, _inputAssetA.erc20Address, _outputAssetA.erc20Address);
+        uint16 currencyId = currencyIds[_inputAssetA.erc20Address];
+        if (currencyId != 0) {
+            if (_inputAssetA.erc20Address == address(0) || _inputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
```

Comment:
Checking for ETH should be done only by asset type because it could happen that the asset would be of type `NOT_USED` and then this condition would pass as well (structs are initialized with 0 values when not specified otherwise)..